### PR TITLE
feat: Simple flag in annotated assignment

### PR DESCRIPTION
**Pull request by dwoznicki**
I'm a bit medium about this solution. It avoids a lookup for parenthesis tokens inside `parse_ann_assign_statement` at the cost of an extra field on the Name expression. I'm not sure if this is a reasonable trade-off.

Failing tests are expected for the time being.

Related: #127

### CODE REVIEW

**Comment by Glyphack** [comment]
Thanks, I would take a look when I'm home. I'd also try to see what https://github.com/astral-sh/ruff does in their parser to detect the "()" on the names.

By the way for tests you can use https://insta.rs/docs/cli/ to update them.

File: parser/src/parser/parser.rs [hunk 1]
```diff
@@ -2934,7 +2945,16 @@ impl<'a> Parser<'a> {
             seen_comma = true;
         }
         if elements.len() == 1 && !seen_comma {
-            return Ok(elements.pop().unwrap());
+            let expr = elements.pop().unwrap();
+            if let Expression::Name(name) = expr {
+                return Ok(Expression::Name(Box::new(Name {
```
Comment (Glyphack) [inline parser/src/parser/parser.rs:2950]: Can't we mutate the name.paranthesized here instead of creating a new one?

**Review by Glyphack — CHANGES_REQUESTED** [review]
I think the change is good. Please add cases to cover this in the snapshots and also update the old snapshots.

Regarding checking the source. I think there are more cases to cover like:

```
(
a
) = 1
```

Where the ( is not exactly the character before the name. So that might introduce more complexity.

**Comment by dwoznicki** [comment]
Okay, I added some tests. I can't help but feel like this is too much overhead for such a niche use case. Perhaps this should be a new expression type, `ParenthesizedName`? How closely are you trying to follow the documented Python AST?

**Comment by Glyphack** [comment]
> Okay, I added some tests. I can't help but feel like this is too much overhead for such a niche use case. Perhaps this should be a new expression type, `ParenthesizedName`? How closely are you trying to follow the standard Python AST node types?

I think this extra field is better than a new node type. Since that new node type has to be handled in type checker when getting type for an expression and might complicate more stuff.

With current approach we can leave the detail inside the parser. I think we can even override fmt debug for the Name to hide this attribute in the debug value so it won't show up in the tests as a change:
https://doc.rust-lang.org/std/fmt/trait.Debug.html#examples

**Comment by dwoznicki** [comment]
Cool, I like that idea. Tests should be good to go. Looks like codecov is getting (badly) throttled.

```
[2024-06-27T06:04:24.041Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 429 - {'detail': ErrorDetail(string='Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 2947s.', code='throttled')}
```

**Pull request merged**